### DISCUSSION TITLE
Empty User Profile Properties lead to invalid PostQuery

### DISCUSSION THREAD

**Discussion by PatrickWeihrauch** [microsoft-search/pnp-modern-search#3446] [General]
**Describe the bug**
In the query template, I want to check for managed properties of pages matching custom user profile properties of the current user.
The user profile properties are multivalue and populated with managed metadata.

I have the following query template:

{|RefinableString120:{User.ProfileProperty1}} OR {|RefinableString121:{User.ProfileProperty2}} OR {|RefinableString122:{User.ProfileProperty3}}

When the user has values in ProfileProperty1 and 2 and 3, the results are shown correctly.

BUT if there are empty values like ProfileProperty1, no results are shown at all, even if the other two properties are filled with valid values which should return results.

The postquery looks like this:
" OR  OR (RefinableString122:Audit) OR (RefinableString122:Brand)"

In this case ProfileProperty3 is filled with "Audit" and "Brand".

It seems this empty ProfileProperties lead to an invalid search query.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Create 3 Site Pages library columns with managed metadata (multivalue).
2. Create pages tagged with values
3. Map Crawled Properties of the columns to 3 RefinableStrings
4.  Create 3 custom User Profile Properties (multivalue, managed metadata)
5. Populate all 3 of them, test the query, then populate only one of them, test the query

**Expected behavior**
All 3 user profile properties populated lead to correct results, 1 or 2 of them empty lead to no results at all, even if the third should lead to results.

**Screenshots**
![image](https://github.com/microsoft-search/pnp-modern-search/assets/156412786/7f0732a5-c62c-4540-91f0-424f02092feb)

**Desktop (please complete the following information):**
 - Browser [Edge]

**Comment by kasperbolarsen** [comment]
Hi Patrick
It sounds like this issue is a general search query issue and not related to PnP Modern Search?

https://techcommunity.microsoft.com/ might be a better place to ask the question :-)

**Comment by wobba** [comment]
If a propery can be empty you should use the notation `{?...}` to remove the part which is empty.

`{?{|RefinableString120:{User.ProfileProperty1}} OR}`

I describe this in my old blog post: https://www.techmikael.com/2014/05/s15e03-query-variables-constant-trouble.html
